src/uucore/src/lib/features/checksum.rs

```diff
@@ -335,35 +299,184 @@ fn determine_regex(
+            }
+        }
+    }
+    Err(ChecksumError::NoProperlyFormattedChecksumLinesFound {
+        filename: get_filename_for_output(filename, input_is_stdin),
+    }
+    .into())
+}
+
+// Converts bytes to a hexadecimal string
+fn bytes_to_hex(bytes: &[u8]) -> String {
+    bytes
+        .iter()
+        .map(|byte| format!("{:02x}", byte))
+        .collect::<Vec<String>>()
+        .join("")
+}
+
+// Gets the expected checksum
+fn get_expected_checksum(
+    filename: &str,
+    caps: &regex::Captures,
+    chosen_regex: &Regex,
+) -> UResult<String> {
+    if chosen_regex.as_str() == ALGO_BASED_REGEX_BASE64 {
+        let ck = caps.name("checksum").unwrap().as_str();
+        match BASE64.decode(ck.as_bytes()) {
+            Ok(decoded_bytes) => {
+                match std::str::from_utf8(&decoded_bytes) {
+                    Ok(decoded_str) => Ok(decoded_str.to_string()),
+                    Err(_) => Ok(bytes_to_hex(&decoded_bytes)), // Handle as raw bytes if not valid UTF-8
+                }
+            }
+            Err(_) => Err(Box::new(
+                ChecksumError::NoProperlyFormattedChecksumLinesFound {
+                    filename: (&filename).to_string(),
+                },
+            )),
         }
+    } else {
+        Ok(caps.name("checksum").unwrap().as_str().to_string())
     }
+}
 
-    Err(
-        ChecksumError::NoProperlyFormattedChecksumLinesFound(get_filename_for_output(
-            filename,
-            input_is_stdin,
-        ))
-        .into(),
-    )
+/// Returns a reader that reads from the specified file, or from stdin if `filename_to_check` is "-".
+fn get_file_to_check(
+    filename_to_check: &str,
+    ignore_missing: bool,
+    res: &mut ChecksumResult,
+) -> Option<Box<dyn Read>> {
+    if filename_to_check == "-" {
+        Some(Box::new(stdin())) // Use stdin if "-" is specified in the checksum file
+    } else {
+        match File::open(filename_to_check) {
+            Ok(f) => {
+                if f.metadata().ok()?.is_dir() {
+                    show!(USimpleError::new(
+                        1,
+                        format!("{}: Is a directory", filename_to_check)
+                    ));
+                    None
+                } else {
+                    Some(Box::new(f))
+                }
+            }
+            Err(err) => {
+                if !ignore_missing {
+                    // yes, we have both stderr and stdout here
+                    show!(err.map_err_context(|| filename_to_check.to_string()));
+                    println!("{}: FAILED open or read", filename_to_check);
+                }
+                res.failed_open_file += 1;
+                // we could not open the file but we want to continue
+                None
+            }
+        }
+    }
+}
+
+/// Returns a reader to the list of checksums
+fn get_input_file(filename_input: &OsStr, input_is_stdin: bool) -> UResult<Box<dyn Read>> {
```

Comment:
`input_is_stdin` seems unnecessary because you can derive from `filename_input` whether the input is stdin. Like it's done in `get_file_to_check()`. And I think you can shorten `filename_input` to `filename`.